### PR TITLE
Expose REACTION_TO_COMPARE and IGNORED_CHANNELS as env variable

### PR DESCRIPTION
**Pull request by Smephite**
Expose `REACTION_TO_COMPARE` and `IGNORED_CHANNELS` as environment variable.

The environment variabels `DISCORD_ALLOWED_REACTION` and `DISCORD_IGNORED_CHANNELS` will be parsed as JSON Array.

If no variable was set, default value from previous version will be used.

Setting an empty array will result in a wildcard.

Also fixes `$$leaderboard` and co. not working in ignored channels.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -4,12 +4,26 @@
...
+NOTIFY_USER = os.environ['NOTIFY_USER']
+IGNORED_CHANNELS = [763798356484161569, 772838189920026635,  839229026194423898] if not 'DISCORD_IGNORED_CHANNELS' in os.environ else json.load(os.environ['DISCORD_IGNORED_CHANNELS'])
+# defaults to General, Lumenauts, Report-spam
+
+
+if not (isinstance(REACTION_TO_COMPARE, list) or  len(REACTION_TO_COMPARE) != 0 and not isinstance(REACTION_TO_COMPARE[0], str)):
+    # REACTION_TO_COMPARE must be str to arr of str; empty array => wildcard 
+    print("DISCORD_ALLOWED_REACTION env variable has to be array of strs!")
+    exit
+if not isinstance(IGNORED_CHANNELS, list) or (len(IGNORED_CHANNELS) != 0 and not isinstance(IGNORED_CHANNELS[0], int)):
```
Comment (MatejMecka) [inline main.py:23]: We leave spaces between the ifs :)

File: main.py [hunk 2]
```diff
@@ -4,12 +4,26 @@
...
 DATABASE_NAME = 'votes.db'
-REACTION_TO_COMPARE = '🐻'
+REACTION_TO_COMPARE = ['🐻'] if not 'DISCORD_ALLOWED_REACTION' in os.environ else json.loads(os.environ['DISCORD_ALLOWED_REACTION'])
 LEADERBOARD_LIMIT = 10
 BOT_TOKEN = os.environ['DISCORD_BOT_TOKEN']
 REQUIRED_ROLE_ID = os.environ['ROLE_ID']
+NOTIFY_USER = os.environ['NOTIFY_USER']
+IGNORED_CHANNELS = [763798356484161569, 772838189920026635,  839229026194423898] if not 'DISCORD_IGNORED_CHANNELS' in os.environ else json.load(os.environ['DISCORD_IGNORED_CHANNELS'])
+# defaults to General, Lumenauts, Report-spam
+
```
Comment (MatejMecka) [inline main.py:17]: There's one extra new line

**Review by MatejMecka — CHANGES_REQUESTED** [review]
Minor style fixes are needed and should be ok

File: main.py (hunk 1 above)
Comment (Smephite) [reply main.py:23]: What exactly do you mean?
Splitting the logic up into nested if-s or new lines where they are logical?

**Review by MatejMecka — APPROVED** [review]
LGTM! Thanks for the PR 😊

**Pull request merged**